From 727826abceea0e3c1ee2050643eb61dba8bfdcd7 Mon Sep 17 00:00:00 2001
From: aiaimimi0920 <153103332+aiaimimi0920@users.noreply.github.com>
Date: Fri, 23 Oct 2026 00:03:45 +0800
Subject: [PATCH] fix(theme): avoid shadowing the upstream CanvasItem window
 member

Keep the inspected Window override behavior unchanged while avoiding the new C4458 warning.

Godot-Patch-Topic: ui.dynamic-color-theme
---
 editor/inspector/editor_inspector.cpp | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/editor/inspector/editor_inspector.cpp b/editor/inspector/editor_inspector.cpp
index cf127adc0b0d5ebf3629e1788177699918dfed2e..f40eba45162472546efa856358ae07e39688226d 100644
--- a/editor/inspector/editor_inspector.cpp
+++ b/editor/inspector/editor_inspector.cpp
@@ -5979,15 +5979,15 @@ void EditorInspector::_property_checked(const String &p_path, bool p_checked) {
 		} else {
 			Variant to_create;
 			Control *control = Object::cast_to<Control>(object);
-			Window *window = Object::cast_to<Window>(object);
+			Window *inspected_window = Object::cast_to<Window>(object);
 			if (control && p_path.begins_with("theme_override_")) {
 				to_create = control->get_used_theme_item(p_path);
-			} else if (window && p_path.begins_with("theme_override_color_schemes/")) {
-				to_create = window->get_theme_color_scheme(p_path.get_slicec('/', 1));
-			} else if (window && p_path.begins_with("theme_override_color_roles/")) {
-				to_create = static_cast<int64_t>(window->get_theme_color_role(p_path.get_slicec('/', 1)));
-			} else if (window && p_path.begins_with("theme_override_colors/")) {
-				to_create = window->get_theme_color(p_path.get_slicec('/', 1));
+			} else if (inspected_window && p_path.begins_with("theme_override_color_schemes/")) {
+				to_create = inspected_window->get_theme_color_scheme(p_path.get_slicec('/', 1));
+			} else if (inspected_window && p_path.begins_with("theme_override_color_roles/")) {
+				to_create = static_cast<int64_t>(inspected_window->get_theme_color_role(p_path.get_slicec('/', 1)));
+			} else if (inspected_window && p_path.begins_with("theme_override_colors/")) {
+				to_create = inspected_window->get_theme_color(p_path.get_slicec('/', 1));
 			} else {
 				List<PropertyInfo> pinfo;
 				object->get_property_list(&pinfo);

base-commit: 5d1cd20f9117e4721fce626454e2323eb9462d0a
prerequisite-patch-id: b38459b409a3326aed65d2433e1341638171f884
prerequisite-patch-id: b205f59ead6c0d8f2c43e71c9f74d4c0d997491c
prerequisite-patch-id: 4dcdddc73066f768cb97c56444abacb21bda061c
prerequisite-patch-id: 34eb3988424d10f34043f09ec0c44ce34db72683
prerequisite-patch-id: 4dd1cec6888d5160c85be2072b9e5dd05411d007
prerequisite-patch-id: 73c65ea8aa25ee3915ec20f693bd27deebfd512e
prerequisite-patch-id: bdb0025d4207a7cf0e1466ca96b2824d238b3caf
prerequisite-patch-id: 8c55a8201fec417bc2629b903ea2d630aedab1d8
prerequisite-patch-id: 9dfaaac9d4a47edf9d1eccce8fa66b69e674a911
prerequisite-patch-id: 17fa223ec74d20c7792e45826a4f0d8ecd7f85d7
prerequisite-patch-id: 0fb9ef780b3f10a7e82968e44defda508734fa91
prerequisite-patch-id: 2af824bf187ff13746bd620b1032f57d166c0114
prerequisite-patch-id: dc8aa901272b27086c9045fc0b9335986d3a8128
prerequisite-patch-id: 0ce30d5045f596dbc6bf72cbcd9bb16472513013
prerequisite-patch-id: 50d820c0fb3ac27ddef8f76b133637558f8bcc8e
prerequisite-patch-id: 69b2baab747aed9362c0ca4c161fc02e5970db44
prerequisite-patch-id: 41b4bd5af73a2a33c2e64048b72e0d4f3c664f19
prerequisite-patch-id: 44c3271900a02cc9b76bea0eeab8318468df3773
prerequisite-patch-id: 14ad80fb2ee81b00b9194186196bba9d350bc28e
prerequisite-patch-id: 67692c41af9b4e8c91298594cc3db77e981863de
prerequisite-patch-id: d5e37dc01427a0639bba6123e6afee995d62ef4a
prerequisite-patch-id: 5b14b52ed17c4439ab6471388936f7b5d8fb4b4a
prerequisite-patch-id: 2e5d11daca8780d00fe2d10677ac0c26098fae5c
prerequisite-patch-id: 06b6906be778d545881f82f2431b2204eb052a6c
prerequisite-patch-id: 6f98206b25188cf56bc55e3fbad4fff5da6d6d4e
prerequisite-patch-id: 9c598b9a701c8d9fb585255c239a2f64c998f3a0
prerequisite-patch-id: 148113e476c059293ce209839c317a4945cc1899
prerequisite-patch-id: 68f8ac60d0f60d3ac0bb1fe23aa79abd660c2009
prerequisite-patch-id: c4af5500ee3eeb05e7e3556ae941b4890b6bc0c2
prerequisite-patch-id: 075c9922d2a9fadb91e8c3402b4e7d2f7bf7f8a1
prerequisite-patch-id: 3b6bc3f484f1654f8bca99626ff2507a0e17a3ca
prerequisite-patch-id: d4fed2df8ba748ad28488271c1b2c42cc88f8c17
prerequisite-patch-id: 6a2c61a480ce572ae1c3692257cf60463d4d3cf9
prerequisite-patch-id: a3cb1ca4de756f71a6ec53eeb037b5ff643baa73
prerequisite-patch-id: 02e0656ba365a7fb2cad0c69ca9ecbcf05ea35f1
prerequisite-patch-id: 6b9471813c5552c60fce6f086172833038fed2ae
prerequisite-patch-id: 9a80815a3f443dabb008cef85de0783e1530a9c9
prerequisite-patch-id: 6cbac35a7fd4756f09c6b54a683203752361d1b8
prerequisite-patch-id: d1179ee94009db93d0dbde68386f42b028303eeb
prerequisite-patch-id: 7fa76bc0dcf10c37ddce6dfe863cf8a15db1a4e3
prerequisite-patch-id: 59285ed00363fbf066365f96211e649c4e9e8707
prerequisite-patch-id: 7fef71675ff2ac560045ba78cb2e69d3d05bf3da
prerequisite-patch-id: cb3f5d7d6c2d28699fd25638d57711baff02bf35
prerequisite-patch-id: 77196bacb24fd150ffc6b82969c4d274c01d6f9a
prerequisite-patch-id: e9942d4f02f72a92af555c51c615991adb95fb8e
prerequisite-patch-id: d4228c26ec2e9e69226ab4b8a1adfe40be71e2e2
prerequisite-patch-id: 433f891f57c7f1058f239694b9e5df4630dfb450
prerequisite-patch-id: 395a24a434c14e0a86c2593175e2c4b2f83d0a15
prerequisite-patch-id: f7b6b4fb0db3aaab3b76e3536517b536e9470ae5
prerequisite-patch-id: 95f67e1b177da895271b995d9c66eec294eee6ed
prerequisite-patch-id: 69e4d9487e53c02a4cf88ebf20b6a44293989b81
prerequisite-patch-id: 162768f49e58e449b45c1a0d2dfe3024495b7cb8
prerequisite-patch-id: 5c94f76c6efe7aa74007ec38c35ac067a423f7c5
prerequisite-patch-id: 7db2bfee60272d5ee87fc909c0135d071c5c92ea
prerequisite-patch-id: b7030a38ad411b6d849e39b2e56870c8b324f0b5
prerequisite-patch-id: df0e2a2f7b3d16733818ba5648bb9eebf64106a9
prerequisite-patch-id: 70d811c3343ae738833b5113d711815bcb671030
prerequisite-patch-id: 9d1d1b183596ae249b9b2fbad41877736a04df46
